src/gt4py/next/program_processors/runners/dace_fieldview/transformations/gpu_utils.py

```diff
@@ -82,171 +86,253 @@ def gt_gpu_transformation(
-        launch_factor=gpu_launch_factor,
-    )
-    return sdfg.apply_transformations_once_everywhere([xform])
-
-
-def _gpu_block_parser(
-    self: GPUSetBlockSize,
-    val: Any,
-) -> None:
-    """Used by the setter of `GPUSetBlockSize.block_size`."""
-    org_val = val
-    if isinstance(val, (tuple | list)):
-        pass
-    elif isinstance(val, str):
-        val = tuple(x.strip() for x in val.split(","))
-    elif isinstance(val, int):
-        val = (val,)
+    for dim in [1, 2, 3]:
+        for arg, val in {
+            "block_size": block_size,
+            "launch_bounds": launch_bounds,
+            "launch_factor": launch_factor,
+        }.items():
+            if f"{arg}_{dim}d" not in kwargs:
+                kwargs[f"{arg}_{dim}d"] = val
+    return sdfg.apply_transformations_once_everywhere(GPUSetBlockSize(**kwargs))
+
+
+def _make_gpu_block_parser_for(
+    dim: int,
+) -> Callable[["GPUSetBlockSize", Any], None]:
+    """Generates a parser for GPU blocks for dimension `dim`.
+
+    The returned function can be used as parser for the `GPUSetBlockSize.block_size_*d`
+    properties.
+    """
+
+    def _gpu_block_parser(
+        self: GPUSetBlockSize,
+        val: Any,
+    ) -> None:
+        """Used by the setter of `GPUSetBlockSize.block_size`."""
+        org_val = val
+        if isinstance(val, (tuple | list)):
+            pass
+        elif isinstance(val, str):
+            val = tuple(x.strip() for x in val.split(","))
+        elif isinstance(val, int):
+            val = (val,)
+        else:
+            raise TypeError(
+                f"Does not know how to transform '{type(org_val).__name__}' into a proper GPU block size."
+            )
+        if len(val) < dim:
+            raise ValueError(
+                f"The passed block size only covers {len(val)} dimensions, but dimension was {dim}."
+            )
+        if 0 < len(val) <= 3:
+            val = [*val, *([1] * (3 - len(val)))]
+        else:
+            raise ValueError(f"Can not parse block size '{org_val}': wrong length")
+        try:
+            val = [int(x) for x in val]
+        except ValueError:
+            raise TypeError(
+                f"Currently only block sizes convertible to int are supported, you passed '{val}'."
+            ) from None
+
+        # Remove over specification.
+        for i in range(dim, 3):
+            val[i] = 1
+        setattr(self, f"_block_size_{dim}d", tuple(val))
+
+    return _gpu_block_parser
+
+
+def _make_gpu_block_getter_for(
+    dim: int,
+) -> Callable[["GPUSetBlockSize"], tuple[int, int, int]]:
+    """Makes the getter for the block size of dimension `dim`."""
+
+    def _gpu_block_getter(
+        self: "GPUSetBlockSize",
+    ) -> tuple[int, int, int]:
+        """Used as getter in the `GPUSetBlockSize.block_size` property."""
+        return getattr(self, f"_block_size_{dim}d")
+
+    return _gpu_block_getter
+
+
+def _gpu_set_block_size_set_launch_bound(
```

Comment:
You are right the name is super confusing.
However, I selected `_gpu_launch_bound_parser` because I wanted the "parser" in it to emphasize what it does.